packages/shared/src/graphql/source/common.ts

```diff
@@ -0,0 +1,23 @@
+import { QueryClient } from 'react-query';
+import { generateQueryKey, RequestKey } from '../../lib/query';
+import { SourceMemberFlag, Squad } from '../sources';
+import { LoggedUser } from '../../lib/user';
+
+export const updateFlagsCache = (
+  client: QueryClient,
+  squad: Squad,
+  user: LoggedUser,
+  update: Partial<SourceMemberFlag>,
+): void => {
+  const queryKey = generateQueryKey(RequestKey.Squad, user, squad?.handle);
+  client.setQueryData<Squad>(queryKey, (oldData) => ({
+    ...oldData,
+    currentMember: {
+      ...oldData.currentMember,
+      flags: {
+        ...(oldData.currentMember.flags ?? {}),
+        ...update,
+      },
+    },
+  }));
```

Comment:
We might wanna check if `oldData` is undefined and then not set anything since query will fetch fresh data for the first time when mounted. Since undefined means it did not mount before.

Reply:
Right. Better to check than break the prod 😅